Objective-C/CBLDocument.mm

```diff
@@ -251,11 +252,17 @@ - (void) replaceC4Doc: (CBLC4Document*)c4doc {
     }
 }
 
+- (void) markAsRemoteDoc {
+    CBL_LOCK(self) {
+        _isRemoteDoc = YES;
```

Comment:
Let throw NSInternalInconsistencyException if it's the doc from the regular Database. I think you can check if the doc has `_c4Doc` or not?

Reply:
Similarly to the regular doc, don't allow the remote doc to be saved in the regular database.